datasetinsights/io/checkpoint.py

```diff
@@ -247,8 +247,7 @@ def load_from_http(estimator, url):
 
     """
     with tempfile.TemporaryDirectory() as temp_dir:
-        path = os.path.join(temp_dir, "estimator_checkpoint")
-        logger.debug(f"Downloading estimator from {url} to {path}")
-        download_file(source_uri=url, dest_path=path)
```

Comment:
Why would this implementation raise `IsADirectoryError` exception?

Reply:
No clue, but previously implementation does not call `download_file` correctly as `<temp_dir>/estimator_checkpoint` is somehow downloaded as a directory. This will fail when estimator tries to load a checkpoint file which points to a directory.